internal/kgateway/krtcollections/policy.go

```
@@ -1210,23 +1216,24 @@ func (h *RoutesIndex) getExtensionRefs(
 	ns string,
 	r []gwv1.HTTPRouteFilter,
 	opts ...ir.PolicyAttachmentOpts,
-) ir.AttachedPolicies {
+) (ir.AttachedPolicies, []error) {
```

Can we return a joined err ere instead? Just more standard instead of making the caller do this. It only makes sense to return a slice if they need to be preserved as separate errors.